includes/RestApi/FlowController.php

```diff
@@ -90,30 +100,23 @@ public function save_onboarding_flow_data( \WP_REST_Request $request ) {
 			);
 		}
 
-		if ( ! ( $flow_data = $this->read_details_from_wp_options() ) ) {
-			  $flow_data              = Flows::get_data();
-			  $flow_data['createdAt'] = time();
-			  // update default data if flow type is ecommerce
-			  $flow_data = $this->update_default_data_for_ecommerce( $flow_data );
-			  $this->save_details_to_wp_options( $flow_data );
-		}
-
-		foreach ( $params as $key => $param ) {
-			if ( $value = $this->array_search_key( $key, $flow_data ) === false ) {
-				return new \WP_Error(
-					'wrong_param_provided',
-					"Wrong Parameter Provided : $key",
-					array( 'status' => 404 )
-				);
-			}
+		$flow_data = $this->get_wp_options_flow_data();
```

Comment:
```suggestion
```
I don't see anywhere this being used after retrieving.

Reply:
It is read and stored in the variable $flow_data, and then used in updating after the post call is made (line 114 )